tasks.py

```diff
@@ -165,12 +171,46 @@ def lint_ruff(context: Context) -> None:
         context.run(exec_cmd)
 
 
+@task
+def lint_markdownlint(context: Context) -> None:
+    """Run markdownlint to check all markdown files."""
+    if not is_tool_installed("markdownlint-cli2"):
+        print(" - markdownlint-cli2 is not installed, skipping documentation linting")
+        return
+
+    print(" - Check documentation with markdownlint-cli2")
+    exec_cmd = "markdownlint-cli2 **/*.{md,mdx} --config .markdownlint.yaml"
+    with context.cd(MAIN_DIRECTORY_PATH):
+        context.run(exec_cmd)
+
+
+@task
+def lint_vale(context: Context) -> None:
+    """Run vale to check all documentation files."""
+    if not is_tool_installed("vale"):
```

Comment:
I'm not sure about these is_tool_installed, it feels like we at some point could get into a situation where we're just running the lint_all command and we move around something so that these tools aren't installed. Then we wouldn't notice it in the pipeline.

Reply:
If someone doesn't have the right tools installed, the pipeline will still catch the issue right now.

These methods are mainly for us to make it easier to validate a branch locally but since not everyone will need them I would prefer to not require everyone to install `vale` & `markdownlint-cli2`